profile_page/database/test_db.py

```diff
@@ -0,0 +1,143 @@
+        print(f"{BColors.OKGREEN}\t[+]\tPass User-Profile database save duplicate.{BColors.ENDC}")
+
+    def tearDown(self):
+        mongo.remove_user(self.user["user_id"])
+
+class DBRead(TestCase):
+    def setUp(self):
+        self.user = {"user_id":"5f7d1b1d8fd2b816c48c148b","badges":[31,24,83],"current story level":9,"email":"ryanb777@umbc.edu","friends":["Kulsoom2","Nick2","Maksim2","Naomi2"],"user_name":"ryan2","password hash":"well,hello there","points":98274,"rank":"diamond","save games":["4(2(3)(no))(6(5))","4(2(3)(1))(6(5))","4(2(3)(1))(6(5))"]}
+        self.fail_phrase = 'nah bro idk about it'
+        mongo.create_user( self.user )
+
+    def test_read_correct(self):
+        """ The user document was read from the database """
+        read_user = mongo.read_one_user( self.user["user_id"] )
+
+        self.assertEqual( read_user["user_id"], self.user["user_id"], msg=f'{BColors.FAIL}\t[-]\tGame Id returned from read board does not equal expected!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass User-Profile database read.{BColors.ENDC}")
+
+    def test_read_nonexist(self):
+        """ The user document should not be read from the database """
+        read_user = mongo.read_one_user( "This name should really not exist in the database, and if it does, YEESH!" )
+
+        self.assertEqual( read_user, self.fail_phrase, msg=f'{BColors.FAIL}\t[-]\tIncorrect return for non-existant user!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass User-Profile database read nonexistant.{BColors.ENDC}")
+
+    def test_read_user_name(self):
+        """ The user's name was read from the database """
+        read_user_name = mongo.read_user_name( self.user["user_id"] )
+
+        self.assertEqual( read_user_name["user_name"], self.user["user_name"], msg=f'{BColors.FAIL}\t[-]\tUsername returned from read board does not equal expected!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass User-Profile database read-name.{BColors.ENDC}")
+
+    def test_read_user_name_nonexist(self):
+        """ The user-name document should not be read from the database """
+        read_user = mongo.read_user_name( "This id should really not exist in the database, and if it does, YEESH!" )
+
+        self.assertEqual( read_user, self.fail_phrase, msg=f'{BColors.FAIL}\t[-]\tIncorrect return for non-existant user id!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass User-Profile database read-name nonexistant.{BColors.ENDC}")
+
+    def tearDown(self):
+        mongo.remove_user(self.user["user_id"])
+
+class DBUpdate(TestCase):
+    def setUp(self):
+        self.user = {"user_id":"5f7d1b1d8fd2b816c48c148b","badges":[31,24,83],"current story level":9,"email":"ryanb777@umbc.edu","friends":["Kulsoom2","Nick2","Maksim2","Naomi2"],"user_name":"ryan2","password hash":"well,hello there","points":98274,"rank":"diamond","save games":["4(2(3)(no))(6(5))","4(2(3)(1))(6(5))","4(2(3)(1))(6(5))"]}
```

Comment:
Node to ourselves: correct the data in the future with what we are using now. But @ryancb4 indicated the fields are gonna say the same so values can be easily replaced.